app/services/config_report_status_calculator.rb

```diff
@@ -1,21 +1,28 @@
 class ConfigReportStatusCalculator
-  # converts a counters hash into a bit field
-  # expects a metrics_to_hash kind of counters
-  # see the report_processor for the implementation
+  # Converts a counters hash into a 64 bit integer (bit field). Required
+  # arguments: counters (hash of counters) or bit_field (raw number for
+  # reverse calculation), metrics (array of expected counter names in
+  # the least significant bit order) and optional size (word size which
+  # is calculated automatically to fit into 64 bit integer result).
   def initialize(options = {})
     @counters   = options[:counters]  || {}
     @raw_status = options[:bit_field] || 0
+    @metrics    = options[:metrics] || ConfigReport::METRIC
+    @size       = options[:size] || (64 / @metrics.size)
+    @max_value  = (1 << @size) - 1
   end
```

Comment:
This class was generalized so it can be used in other contexts (e.g. new report or openscap report).